Support catch2 v3

Catch2 testing framework has been working for some time now on a new, [major version (v3.0.1)](https://github.com/catchorg/Catch2/blob/devel/docs/release-notes.md#301) which breaks the current fakeit build for catch2.

On the most basic level, the header files names had been changed. Trying to compile fakeit+catch2 v3 will output:
```
[build] unittests_poc\tests_catch\fakeit.hpp(1137,13): fatal error C1083: Cannot open include file: 'catch.hpp': No such file or directory [unittests_poc\build\tests_catch\testlib.vcxproj]
[build] Build finished with exit code 1
```

--- a/config/catch/CatchFakeit.hpp
+++ b/config/catch/CatchFakeit.hpp
@@ -8,6 +8,8 @@
 #elif __has_include("catch2/catch_all.hpp")
 #   include <catch2/catch_assertion_result.hpp>
 #   include <catch2/catch_test_macros.hpp>
+#elif __has_include("catch_amalgamated.hpp")
+#   include <catch_amalgamated.hpp>
 #else
 #   include <catch.hpp>
 #endif

--- a/config/catch/CatchFakeit.hpp
+++ b/config/catch/CatchFakeit.hpp
@@ -5,6 +5,9 @@
 #include "mockutils/to_string.hpp"
 #if __has_include("catch2/catch.hpp")
 #   include <catch2/catch.hpp>
+#elif __has_include("catch2/catch_all.hpp")
+#   include <catch2/catch_assertion_result.hpp>
+#   include <catch2/catch_test_macros.hpp>
 #else
 #   include <catch.hpp>
 #endif

--- a/single_header/catch/fakeit.hpp
+++ b/single_header/catch/fakeit.hpp
@@ -2,7 +2,7 @@
 /*
  *  FakeIt - A Simplified C++ Mocking Framework
  *  Copyright (c) Eran Pe'er 2013
- *  Generated: 2021-05-12 13:47:04.979584
+ *  Generated: 2022-02-25 00:22:22.032093
  *  Distributed under the MIT License. Please refer to the LICENSE file at:
  *  https://github.com/eranpeer/FakeIt
  */
@@ -1133,6 +1133,9 @@ namespace fakeit {
 }
 #if __has_include("catch2/catch.hpp")
 #   include <catch2/catch.hpp>
+#elif __has_include("catch2/catch_all.hpp")
+#   include <catch2/catch_assertion_result.hpp>
+#   include <catch2/catch_test_macros.hpp>
 #else
 #   include <catch.hpp>
 #endif

--- a/single_header/catch/fakeit.hpp
+++ b/single_header/catch/fakeit.hpp
@@ -2,7 +2,7 @@
 /*
  *  FakeIt - A Simplified C++ Mocking Framework
  *  Copyright (c) Eran Pe'er 2013
- *  Generated: 2022-02-25 00:22:22.032093
+ *  Generated: 2022-02-25 00:37:04.377858
  *  Distributed under the MIT License. Please refer to the LICENSE file at:
  *  https://github.com/eranpeer/FakeIt
  */
@@ -1136,6 +1136,8 @@ namespace fakeit {
 #elif __has_include("catch2/catch_all.hpp")
 #   include <catch2/catch_assertion_result.hpp>
 #   include <catch2/catch_test_macros.hpp>
+#elif __has_include("catch_amalgamated.hpp")
+#   include <catch_amalgamated.hpp>
 #else
 #   include <catch.hpp>
 #endif

